### PR TITLE
Map zoom again

### PR DESCRIPTION
**Pull request by daniel-sousa-me**
This time it interacts well with Tile/TokenSelector

### CODE REVIEW

File: assets/app/view/game/hex.rb [hunk 1]
```diff
@@ -94,7 +94,9 @@ def transform
       def on_hex_click
         return if @actions.empty? && @role != :tile_page
 
-        return store(:tile_selector, nil) if !@clickable || (@hex == @tile_selector&.hex && !@tile_selector.tile)
+        if !@clickable || (@hex == @tile_selector&.hex && !(@tile_selector.respond_to?(:tile) && @tile_selector.tile))
```
Comment (tobymao) [inline assets/app/view/game/hex.rb:97]: why do you call respond_to?

File: assets/app/view/game/map_controls.rb [hunk 2]
```diff
@@ -5,15 +5,16 @@
 module View
   module Game
     class MapControls < Snabberb::Component
-      needs :show_coords, default: true, store: true
-      needs :show_location_names, default: true, store: true
-      needs :show_starting_map, default: false, store: true
+      needs :show_coords, store: true
```
Comment (tobymao) [inline assets/app/view/game/map_controls.rb:8]: set the default to be nil

File: assets/app/view/game/hex.rb (hunk 1 above)
Comment (daniel-sousa-me) [reply assets/app/view/game/hex.rb:97]: It's a small bug I found while testing this. The `@tile_selector` variable can actually be a `TokenSelector` and this was throwing an error

File: assets/app/view/game/map_controls.rb (hunk 2 above)
Comment (michaeljb) [reply assets/app/view/game/map_controls.rb]: 9bb22f3e

**Comment by michaeljb** [comment]
@tobymao @daniel-sousa-me I took the liberty of fixing the merge conflicts, and I also set the `shoo_coords` default to `nil` everywhere. Is there anything else that this PR needs before merging?

**Comment by daniel-sousa-me** [comment]
Thank you very much! I thought I had already taken care of the default :/

**Comment by michaeljb** [comment]
No worries :slightly_smiling_face:

**Review by tobymao — APPROVED** [review]

**Pull request merged**